core/src/main/resources/org/wildfly/channel/v2.0.0/schema.json

```diff
@@ -0,0 +1,146 @@
+      "description": "Vendor of the channel.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "Name of the vendor",
+          "type": "string"
+        },
+        "support": {
+          "description": "Support level provided by the vendor",
+          "type": "string",
+          "enum": [
+            "supported",
+            "tech-preview",
+            "community"
+          ]
+        }
+      },
+      "required": ["name", "support"]
+    },
+    "requires": {
+      "description": "Channels that are required by this channel.",
+      "type": "array",
+      "items": {
+        "type": "object",
+        "minItems": 1,
+        "properties": {
+          "groupId": {
+            "description": "GroupID Maven coordinate of the channel",
+            "type": "string"
+          },
+          "artifactId": {
+            "description": "ArtifactID Maven coordinate of the channel",
+            "type": "string"
+          },
+          "version": {
+            "description": "Version Maven coordinate of the channel",
+            "type": "string"
+          }
+        },
+        "required": ["groupId", "artifactId"]
+      }
+    },
+    "repositories": {
+      "description": "Repositories the channel uses to resolve it's streams.",
+      "type": "array",
+      "minItems": 1,
+      "items": {
+        "type": "object",
+        "properties": {
+          "id": {
+            "description": "Id of the repository",
+            "type": "string"
+          },
+          "url": {
+            "description": "URL of the repository",
+            "type": "string"
+          }
+        },
+        "required": ["id", "url"]
+      }
+    },
+    "manifest": {
+      "description": "Location of the channel's manifest",
+      "type": "object",
+      "properties": {
+        "gav": {
+          "description": "Maven coordinates of the manifest in format of 'groupId:artifactId[:version]'",
+          "type": "string"
+        },
+        "url": {
+          "description": "Static URL of the manifest file.",
+          "type": "string"
+        }
+      },
+      "oneOf": [
+        {
+          "required": [
+            "gav"
+          ]
+        },
+        {
+          "required": [
+            "url"
+          ]
+        }
+      ]
+    },
+    "blocklist": {
+      "description": "Location of the channel's blocklist",
+      "type": "object",
+      "properties": {
+        "gav": {
+          "description": "Maven coordinates of the blocklist in format of 'groupId:artifactId[:version]'",
+          "type": "string"
+        },
+        "url": {
+          "description": "Static URL of the blocklist file.",
+          "type": "string"
+        }
+      },
+      "oneOf": [
+        {
+          "required": [
+            "gav"
+          ]
+        },
+        {
+          "required": [
+            "url"
+          ]
+        }
+      ]
+    },
+    "resolves-if-no-stream": {
```

Comment:
in the final version, we should maybe add a link to the spec to describe the semantics of the enumeration

Reply:
is there a default strategy?

Reply:
I would say that the default strategy should be `none` - if the artifact is not defined in the streams, throw an error. That feels like the natural option at least.